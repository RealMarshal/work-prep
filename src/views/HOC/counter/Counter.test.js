import React, { Fragment } from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import withCounter from './Counter'

const CounterView = ({value, inc, dec}) => (
  <Fragment>
    <p>{ value }</p>
    <button onClick={inc}>inc</button>
    <button onClick={dec}>dec</button>
  </Fragment>
)

const Counter = withCounter(CounterView)

describe('CounterHOC', () => {
  const rendered = mount(<Counter />)
  let { value, inc, dec } = rendered.find(CounterView).props()

  it('should render passed component', () => {
    expect(toJson(rendered)).toMatchSnapshot()
  })

  it('should pass initial value to the component', () => {
    expect(value).toEqual(0)
  })

  it('should increment value when inc is called', () => {
    inc()
    rendered.update()
    const { value: newValue, setValue } = rendered.find(CounterView).props()
    expect(newValue).toEqual(value + 1)
    setValue(0)
  })

  it('should decrement value when dec is called', () => {
    dec()
    rendered.update()
    const { value: newValue, setValue } = rendered.find(CounterView).props()
    expect(newValue).toEqual(value - 1)
    setValue(0)
  })
})