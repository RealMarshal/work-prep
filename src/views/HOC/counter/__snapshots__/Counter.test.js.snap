// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterHOC should render passed component 1`] = `
<withState(withHandlers(Counter))>
  <withHandlers(Counter)
    setValue={[Function]}
    value={0}
  >
    <Counter
      dec={[Function]}
      inc={[Function]}
      setValue={[Function]}
      value={0}
    >
      <p>
        0
      </p>
      <button
        onClick={[Function]}
      >
        inc
      </button>
      <button
        onClick={[Function]}
      >
        dec
      </button>
    </Counter>
  </withHandlers(Counter)>
</withState(withHandlers(Counter))>
`;
